internal/version/service.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package version
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/sync2"
+)
+
+// Config contains the necessary Information to check the Software Version
+type Config struct {
+	ServerAddress  string        `help:"server address to check its version against" default:"https://satellite.stefan-benten.de/version"`
```

Comment:
This needs to be changes still. @jtolds 
I have this server running for testing

Reply:
k